impls/jq/step5_tco.jq

```diff
@@ -261,27 +238,29 @@ def EVAL(env):
                                 TCOWrap($_menv; $_orig_retenv; false)
                         )
             ) //
-            (select(.kind == "vector") |
-                if .value|length == 0 then
-                    {
-                        kind: "vector",
-                        value: []
-                    } | TCOWrap($_menv; $_orig_retenv; false)
-                else
-                    [ { env: $_menv, list: .value } | map_with_env ] as $res |
-                    {
-                        kind: "vector",
-                        value: $res | map(.value)
-                    } | TCOWrap($res | last.env; $_orig_retenv; false)
-                end
+            (
+                select(.kind == "vector") |
+                .value |
+                reduce .[] as $x ({expr:[], env:env};
```

Comment:
The issue with REGRESS is simple enough, this should be `$_menv` (env is the initial env, which is why you get "b not found" in the regression test), you use the right value for later steps.